subset/switches/src/main/java/switchtest/SwitchInterrogator.java

```diff
@@ -218,99 +229,111 @@ public void receiveData(String data) {
   }
 
   private void parse_packet(String raw_data, String[] show_expected, int[] show_pointers) {
-    int start = 0;
-    for (int port = 0; port < number_switch_ports; port++) {
-      for (int x = 0; x < show_expected.length; x++) {
-        start = recursive_data(raw_data, show_expected[x], start);
-        show_pointers[x] = start;
-      }
+    try {
+      int start = 0;
+      for (int port = 0; port < number_switch_ports; port++) {
+        for (int x = 0; x < show_expected.length; x++) {
+          start = recursive_data(raw_data, show_expected[x], start);
+          show_pointers[x] = start;
+        }
 
-      int chunk_s = show_pointers[0];
-      int chunk_e = show_pointers[3];
+        int chunk_s = show_pointers[0];
+        int chunk_e = show_pointers[3];
 
-      if (chunk_e == -1) chunk_e = raw_data.length();
+        if (chunk_e == -1) chunk_e = raw_data.length();
 
-      String temp_data = raw_data.substring(chunk_s, chunk_e);
+        String temp_data = raw_data.substring(chunk_s, chunk_e);
 
-      if (debug) System.out.println("length" + temp_data.length() + "temp_data:" + temp_data);
+        if (debug) System.out.println("length" + temp_data.length() + "temp_data:" + temp_data);
 
-      if (requestFlag == (interfacePos + 1)) {
-        parse_single(
-            temp_data, show_interface_expected, show_interface_pointers, show_interface_data);
-      } else if (requestFlag == (platformPos + 1)) {
-        parse_single(temp_data, show_platform_expected, show_platform_pointers, show_platform_data);
+        if (requestFlag == (interfacePos + 1)) {
+          parse_single(
+              temp_data, show_interface_expected, show_interface_pointers, show_interface_data);
+        } else if (requestFlag == (platformPos + 1)) {
+          parse_single(
+              temp_data, show_platform_expected, show_platform_pointers, show_platform_data);
+        }
       }
+    } catch (Exception e) {
+      System.err.println("Exception parse_packet:" + e.getMessage());
```

Comment:
not the right behavior (just continuing to execute after exception)

Reply:
only useful when debugging the code during development, 
shouldnt happen in a real world